lib/server.py

```diff
@@ -88,25 +93,30 @@ def __init__(self, ini, test_suite=None):
     def prepare_args(self, args=[]):
         return args
 
-    def cleanup(self, full=False):
-        if full:
-            shutil.rmtree(self.vardir)
-            return
-        for re in self.re_vardir_cleanup:
-            for f in glob.glob(os.path.join(self.vardir, re)):
+    def pre_cleanup(self):
+        if self.ini['pre_cleanup']:
+            self.cleanup()
```

Comment:
Beside cleanup of that server we need to cleanup xlog/snap of non-default servers were run within previous test. We don't know non-default server names, so need to either save them somewhere or guess them from directory names inside vardir.

Reply:
Sergei found that we should not clean non-default servers in pre_cleanup, because it already cleaned at `create server`.